### PR TITLE
[pvr][fix] wrong selected channel item in PVR windows

### PR DESCRIPTION
**Pull request by xhaggi**
This is a follow-up for #8582 and will fix the root cause of the issue reported by @MartijnKaijser.

@ksooo after your refactoring of the guide window, where you moved all the stuff from `Update()` to `GetDirectory()`, you forget to check which stuff needs to be executed after the items are assigned to the view control. This is the reason why the first channel is selected at the first access because `epgGridContainer->SetChannel()` has no effect if called before the items are populated to the control. This will be fixed by the second commit.

I've also moved the `SET_CONTROL_LABEL` stuff from `GetDirectory()` to `UpdateButtons()` where this stuff should be life in.

### CODE REVIEW

File: xbmc/pvr/windows/GUIWindowPVRBase.cpp [hunk 1]
```diff
@@ -79,12 +79,7 @@ void CGUIWindowPVRBase::SetSelectedItemPath(bool bRadio, const std::string &path
 
 std::string CGUIWindowPVRBase::GetSelectedItemPath(bool bRadio)
 {
-  if (!m_selectedItemPaths.at(bRadio).empty())
-    return m_selectedItemPaths.at(bRadio);
-  else if (g_PVRManager.IsPlaying())
-    return g_application.CurrentFile();
-
-  return "";
+  return m_selectedItemPaths.at(bRadio);
```
Comment (xhaggi) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: As we call `SetSelectedItemPath()` before we tune to a channel, `m_selectedItemPaths will always hold a valid channel path and we don't need to check `g_PVRManager.IsPlaying()` here.

**Comment by xhaggi** [comment]
@Jalle19 mind taking a look too.

File: xbmc/pvr/windows/GUIWindowPVRBase.cpp [hunk 2]
```diff
@@ -838,7 +833,8 @@ void CGUIWindowPVRBase::UpdateButtons(void)
 
 void CGUIWindowPVRBase::UpdateSelectedItemPath()
 {
-  m_selectedItemPaths.at(m_bRadio) = m_viewControl.GetSelectedItemPath();
+  if (!m_viewControl.GetSelectedItemPath().empty())
```
Comment (xhaggi) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: `UpdateSelectedItemPath()` is called several times without a valid `m_viewControl.GetSelectedItemPath()`. Thats the reason why sometimes the wrong channel (the first item) is selected.

**Comment by xhaggi** [comment]
jenkins build this please

**Comment by ksooo** [comment]
Code changes are looking good to me.

**Comment by xhaggi** [comment]
any objections if not I'll merge it in tomorrow.

**Comment by ksooo** [comment]
Go ahead.

**Pull request merged**